model/DataBaseAccess.php

```diff
@@ -177,7 +178,28 @@ public function addMultiplePermissions(array $permissionData)
                 }
             }
         }
-        $this->getPersistence()->insertMultiple(self::TABLE_PRIVILEGES_NAME, $insert);
+
+        $this->getLogger()->debug(
```

Comment:
Moved new code into a separate method, and wrapped all insertions with a single transactional() call in 60a7e8314c5ae87087cd8b9e659beeae2886ad7e. I totally missed about the possibility of wrapping the insertions into a transaction, thanks!

I'll update unit tests next.

Reply:
Tests updated in 9694d05fb318aa5398fe392ee9a5b1150d74e5e2 (then, 44efff00c386b731c528f51f1e101cd1c5cbb9fa removes the additional echo I accidentally committed).

As these tests are checking edge cases with sets of more than 20K permissions, running them is slow and takes a lot of memory. @gabrielfs7 @shpran wdyt about setting `INSERT_CHUNK_SIZE` as a parameter of `DataBaseAccess` instead of a constant? We may allow passing it as an option to the constructor (inherited from the `Configurable` class), so it is set to 20K by default but initialized to some lower value for tests.

Reply:
We've agreed to replace the `INSERT_CHUNK_SIZE` class constant with a variable so we can lower the number of rows per query for unit tests using a  setter method. The new variable will have a default value of  20K rows.

Reply:
Changed here :point_right: b81293facb914a03a0e3f2e401e6abc58897da0f

Reply:
New code has been moved to a private method, queries now run inside a transaction, and we've already updated unit tests and simplified them (i.e. eb507d0e7496970760cd09c3deb10c3dbce8e17e). Therefore, closing this thread as I think we've already handled all required points for it.